plugins/catalog-backend-module-github/src/service/GitHubEntityProviderCatalogModule.ts

```diff
@@ -51,16 +41,10 @@ export const githubEntityProviderCatalogModule = createBackendModule({
         scheduler: schedulerServiceRef,
       },
       async init({ config, catalog, logger, scheduler }) {
-        const scheduleDef = options?.schedule ?? {
-          frequency: { seconds: 600 },
-          timeout: { seconds: 900 },
-          initialDelay: { seconds: 3 },
-        };
```

Comment:
This sparked an interesting discussion internally. The duality of what goes in config vs what goes in code, and what that means for classes that are code heavy - say, if a module wants a filter function passed in. Whether the module right in this file should be seen as a "convenience"/quick-start module and that you'll make your own module instead of you want to pass in more advanced options. And whether perhaps it would make sense to allow people to install the same module multiple times with different settings in some situations...

Just thinking out loud, no action on your part :)

Reply:
all those thoughts make sense to me. A good example is likely the msgraph module where you can pass transformations to influence/participate in the process of creating User and Group entities.

Reply:
@freben I realized that modules cannot have extension points (by design)

Maybe it makes sense provide it though. This would allow to add your own module which participates in the creation of the module and e.g., provides the filter functions (and sets up whatever is needed for that).

This would remove the need to copy paste + modify it for the "convenience" variant.

Re multiple instances:
Probably, I would expect them to be installed just once, however inside of them they should take care of creating multiple instances of whatever they provide if this makes sense.
And example can be some of the catalog providers which actually create multiple instances of their EntityProvider implementation -- in their case driven by the config. If you have one provide config, there will be one. If you have multiple configs, there will be multiple instances.